applications/nrf5340_audio/src/modules/sd_card.c

```diff
@@ -180,6 +190,126 @@ int sd_card_read(char const *const filename, char *const data, size_t *size)
 	return 0;
 }
 
+int sd_card_segment_open(char const *const filename, char const *const path_to_file)
+{
+	int ret;
+
+	if (strlen(path_to_file) > PATH_MAX_LEN) {
+		LOG_ERR("Filepath is too long");
+		return -EINVAL;
+	}
+
+	char abs_path_name[PATH_MAX_LEN + 1] = SD_ROOT_PATH;
+	strcat(abs_path_name, path_to_file);
+	LOG_INF("abs path name:\t%s\n", abs_path_name);
+
+	if (!sd_init_success) {
+		return -ENODEV;
+	}
+
+	if (strlen(filename) > CONFIG_FS_FATFS_MAX_LFN) {
+		LOG_ERR("Filename is too long");
+		return -ENAMETOOLONG;
+	}
+
+	strcat(abs_path_name, filename);
+	fs_file_t_init(&f_seg_read_entry);
+
+	ret = fs_open(&f_seg_read_entry, abs_path_name, FS_O_READ);
+	if (ret) {
+		LOG_ERR("Open file failed");
+		return ret;
+	}
+
+	seg_read_started = true;
+
+	return 0;
+}
+
+int sd_card_segment_read(char *const data, size_t *size)
+{
+	int ret;
+
+	if (!seg_read_started) {
+		return -EBUSY;
+	}
+
+	ret = fs_read(&f_seg_read_entry, data, *size);
+	if (ret < 0) {
+		LOG_ERR("Read file failed");
+		return ret;
+	}
+
+	*size = ret;
+
+	return 0;
+}
+
+int sd_card_segment_peek(char *const data, size_t *size)
+{
+	int ret;
+
+	off_t offset;
+
+	if (!seg_read_started) {
+		return -EBUSY;
+	}
+
+	offset = fs_tell(&f_seg_read_entry);
+	if (offset < 0) {
+		LOG_ERR("Fs tell failed");
+		return offset;
+	}
+	ret = fs_read(&f_seg_read_entry, data, *size);
+	if (ret < 0) {
+		LOG_ERR("Read file failed");
+		return ret;
+	}
+	ret = fs_seek(&f_seg_read_entry, offset, 0);
+	if (ret < 0) {
+		LOG_ERR("Fs seek failed");
+		return ret;
+	}
+	*size = ret;
+
+	return 0;
+}
+
+int sd_card_segment_skip(const size_t *size)
+{
+	int ret;
+
+	if (!seg_read_started) {
+		return -EBUSY;
+	}
+	ret = fs_seek(&f_seg_read_entry, *size, FS_SEEK_CUR);
+	if (ret < 0) {
+		LOG_ERR("Fs seek failed. Return value: %d\n", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+int sd_card_segment_close(void)
+{
+	int ret;
+
+	if (!seg_read_started) {
+		return -EBUSY;
+	}
+
+	ret = fs_close(&f_seg_read_entry);
+	if (ret) {
+		LOG_ERR("Close file failed");
```

Comment:
Please add return value in the prints:
```suggestion
		LOG_ERR("Close file failed: %d", ret);
```

Goes for all the error prints in this file